### PR TITLE
WIP - initial commit

### PR DESCRIPTION
**Pull request by fredsmith**

### CODE REVIEW

File: gcd.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#! /usr/bin/env python3
+
+
+def gcd(args):
```
Comment (feanil) [inline gcd.py]: Is this different from https://stackoverflow.com/questions/11175131/code-for-greatest-common-divisor-in-python ?  If so, add some comments.

File: hermes.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+@click.option('--debug/--no-debug', default=False)
+@click.option('--config', '-c', type=(str, str, str, int), help='filename url update_command seconds_between_updates', multiple=True)
+@click.option('--yamlfile', '-y', type=click.Path())
+
+def watch_config(config, daemon, debug, yamlfile):
+    if yamlfile:
+        with open(yamlfile) as yamlhandle:
+            config = yaml.load(yamlhandle)
+    seconds = 0
+    update_interval = gcd.gcd([update_intervals[3] for update_intervals in config])
```
Comment (feanil) [inline hermes.py]: This all seems un-necassarily fancy, why not just one update interval for the script?

File: hermes.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    if debug:
+        sys.stdout.write('DEBUG: Sleep Interval %d\n' % update_interval)
+    file_timestamps = {}
+
+    while True:
+        for config_file in config:
```
Comment (feanil) [inline hermes.py]: Do you mean a config block?  Can you provide a sample of what the yaml file looks like?

**Review by feanil — COMMENTED** [review]
In general you're using a lot of lists where I think dictionaries would be more clear.
It's harder to know what `config_file[0]` is than to know what `config_file["file_name"]` is.

**Review by fredsmith — COMMENTED** [review]
Discussion suggestions:
  - move to 3 arguments, instead of --config positional argument.  --file, --url, --command. check that length is the same, then Iterate through arguments on cli, building list of dicts.
  - move to a single update interval, factor out GCD logic.
  - change return types to True/False.  Make the exception case to not download.

**Review by coryleeio — APPROVED** [review]

**Pull request merged**